Steps to start Autogen

uv init

uv venv

source .venv/Scripts/activate

uv add autogen-agentchat "autogen-ext[openai]" ipykernel python-dotenv asyncio autogen-core

In [1]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
import os

load_dotenv()
api_key = os.getenv("GEMINI_API_KEY")
# print(api_key)

In [2]:
model_client = OpenAIChatCompletionClient(model="gemini-2.5-flash", api_key= api_key)

In [6]:
assistant = AssistantAgent(name = "assistant", model_client = model_client)

In [7]:
result = await assistant.run(task= "Whats the capital of India and tell me something more about India ?")

In [8]:
print(result)

messages=[TextMessage(id='55a3c6fb-1f19-4dac-ad63-e3fa087d858c', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 20, 8, 37, 7, 413278, tzinfo=datetime.timezone.utc), content='Whats the capital of India and tell me something more about India ?', type='TextMessage'), TextMessage(id='d333217c-2109-4e5c-a3be-a54c8b115464', source='assistant', models_usage=RequestUsage(prompt_tokens=39, completion_tokens=109), metadata={}, created_at=datetime.datetime(2026, 2, 20, 8, 37, 10, 544443, tzinfo=datetime.timezone.utc), content="The capital of India is **New Delhi**.\n\nHere's something more about India:\n\nIndia is the **world's most populous country** and the **seventh-largest country by area**. It is incredibly diverse, with 22 official languages, a rich tapestry of cultures, religions, and traditions, and a wide variety of geographical features ranging from the Himalayas in the north to tropical coastlines in the south. India is also known as the birthplace

messages=[TextMessage(id='55a3c6fb-1f19-4dac-ad63-e3fa087d858c', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 2, 20, 8, 37, 7, 413278, tzinfo=datetime.timezone.utc), content='Whats the capital of India and tell me something more about India ?', type='TextMessage'), TextMessage(id='d333217c-2109-4e5c-a3be-a54c8b115464', source='assistant', models_usage=RequestUsage(prompt_tokens=39, completion_tokens=109), metadata={}, created_at=datetime.datetime(2026, 2, 20, 8, 37, 10, 544443, tzinfo=datetime.timezone.utc), content="The capital of India is **New Delhi**.\n\nHere's something more about India:\n\nIndia is the **world's most populous country** and the **seventh-largest country by area**. It is incredibly diverse, with 22 official languages, a rich tapestry of cultures, religions, and traditions, and a wide variety of geographical features ranging from the Himalayas in the north to tropical coastlines in the south. India is also known as the birthplace of four major world religions: Hinduism, Buddhism, Jainism, and Sikhism.", type='TextMessage')] stop_reason=None

In [10]:
print(result.messages[-1].content)

The capital of India is **New Delhi**.

Here's something more about India:

India is the **world's most populous country** and the **seventh-largest country by area**. It is incredibly diverse, with 22 official languages, a rich tapestry of cultures, religions, and traditions, and a wide variety of geographical features ranging from the Himalayas in the north to tropical coastlines in the south. India is also known as the birthplace of four major world religions: Hinduism, Buddhism, Jainism, and Sikhism.
